### DISCUSSION TITLE
Override entities and ids of createEntityAdapter?

### DISCUSSION THREAD

**Discussion by mariolim96** [reduxjs/redux-toolkit#2755] [Q&A]
It is possible? 
how can I do it?

**Comment by markerikson** [comment]
@mariolim96 : this is a really vague question :(

What are you trying to accomplish, and why?

Can you show some code for what you want to do?

**Reply by mariolim96** [reply]
usually, after I call getInitialState of a createEntityAdapter I receive a state like this :  
```
{
    ids: [],
    entities: {},
}

```
but I would have a state like : 
```
{
    documentIds: [],
    documents: {},
}
```
or like this:
```
{
 document: {
     ids: [],
    entities: {},
  }
}
```

**Reply by markerikson** [reply]
You definitely cannot change the names from `ids` to `documentIds`.

You _can_ use the entity adapter for a nested field:

```ts
const initialState = {
  document: someAdapter.getInitialState()
}
```

but you need to then pass the right field into the adapter yourself:

```ts
createSlice({
  name,
  initialState,
  reducers: {
    someReducer(state, action) {
      someAdapter.addOne(state.document, action)
    }
  }
})
```
